mobilecoind/src/payments.rs

```diff
@@ -1078,62 +1232,91 @@ impl<T: BlockchainConnection + UserTxConnection + 'static, FPR: FogPubkeyResolve
                 )
                 .map_err(|_| Error::TxBuild("failed creating InputCredentials".into()))?,
             );
+            *balance_sheet.entry(utxo.token_id.into()).or_default() -= utxo.value as i128;
+        }
+
+        for (sci_for_tx, sci_amounts) in scis {
+            if sci_for_tx.partial_fill_value == 0 {
+                tx_builder.add_presigned_input(sci_for_tx.sci.clone())?;
+            } else {
+                // The tx_builder expects to be given the partial fill change value,
```

Comment:
Maybe it makes sense to change the transaction builder api (in a separate pr)?
My thinking is that since this API is always used by the claimer of an SCI, wouldn't it make sense that they always specify how much they want to get? I think they'll always start with that so they'll end up doing this exact subtraction every time this is used.

Reply:
gonna make an issue

Reply:
#3230